### PR TITLE
Text index approaches

### PR DESCRIPTION
**Pull request by evanchooly**
implementing text indexing in two different fashions.  let the voting commence.

### CODE REVIEW

File: morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java [hunk 1]
```diff
@@ -384,6 +390,102 @@ private void processClassAnnotations(final DBCollection dbColl, final MappedClas
                 }
             }
         }
+        
+        mapTextIndexed(dbColl, mc);
```
Comment (evanchooly) [inline morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java]: These two method calls are the respective hooks for each approach

File: morphia/src/test/java/org/mongodb/morphia/indexes/TestTextIndexing.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import org.mongodb.morphia.annotations.Id;
+import org.mongodb.morphia.annotations.Language;
+import org.mongodb.morphia.annotations.Text;
+import org.mongodb.morphia.annotations.TextIndex;
+import org.mongodb.morphia.annotations.TextIndexed;
+
+import java.util.List;
+
+public class TestTextIndexing extends TestBase {
+    @Entity
```
Comment (evanchooly) [inline morphia/src/test/java/org/mongodb/morphia/indexes/TestTextIndexing.java]: First approach:  multiple annotations

File: morphia/src/test/java/org/mongodb/morphia/indexes/TestTextIndexing.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        @Text
+        private String name;
+        @Text(10)
+        private String nickName;
+        @Language
+        private String nativeTongue;
+    }
+
+    @Entity
+    @TextIndex(name = "singleAnnotation", language = "russian", value = {"name", "nickName"}, weights = {"nickName : 10"},
```
Comment (evanchooly) [inline morphia/src/test/java/org/mongodb/morphia/indexes/TestTextIndexing.java]: second approach one uber-annotation

**Comment by evanchooly** [comment]
Here are two approaches to defining text indexes:  the first uses 3 annotations; the second only one.  I lean slightly toward the first approach as it's a bit easier to parse and verify the values.  The second on is definitely the one stop shop but requires a bit of duplication between the 'value' and 'weights' parameters and then there's that ugly string parsing needed to get at the weight values.  Since it's a string there's no way to enforce that only int values are passed in until validated at runtime.

Still, here are two options.  Which feels better?

**Comment by trishagee** [comment]
The first is arguably uglier, but it does allow much greater flexibility for extension. And given we don't know what sort of indexes are coming in the future, it feels a bit safer.

**Pull request closed without merging**